app/src/main/java/com/azuresamples/msalnativeauthandroidkotlinsampleapp/MFAFragment.kt

```diff
@@ -0,0 +1,236 @@
+    private var _binding: FragmentEmailPasswordBinding? = null
+    private val binding get() = _binding!!
+
+    companion object {
+        private val TAG = MFAFragment::class.java.simpleName
+        private enum class STATUS { SignedIn, SignedOut }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        _binding = FragmentEmailPasswordBinding.inflate(inflater, container, false)
+        val view = binding.root
+
+        (activity as? AppCompatActivity)?.supportActionBar?.title = getString(R.string.title_email_otp_mfa)
+
+        authClient = AuthClient.getAuthClient()
+
+        init()
+
+        return view
+    }
+
+    override fun onResume() {
+        super.onResume()
+        getStateAndUpdateUI()
+    }
+
+    private fun init() {
+        binding.signUp.visibility = View.GONE
+
+        initializeButtonListeners()
+    }
+
+    private fun initializeButtonListeners() {
+        binding.signIn.setOnClickListener {
+            signIn()
+        }
+
+        binding.signOut.setOnClickListener {
+            signOut()
+        }
+    }
+
+    private fun getStateAndUpdateUI() {
+        CoroutineScope(Dispatchers.Main).launch {
+            val accountResult = authClient.getCurrentAccount()
+            when (accountResult) {
+                is GetAccountResult.AccountFound -> {
+                    displaySignedInState(accountResult.resultValue)
+                }
+                is GetAccountResult.NoAccountFound -> {
+                    displaySignedOutState()
+                }
+                is GetAccountError -> {
+                    displayDialog(getString(R.string.msal_exception_title), accountResult.errorMessage)
+                }
+            }
+        }
+    }
+
+    private fun signIn() {
+        CoroutineScope(Dispatchers.Main).launch {
+            val email = binding.emailText.text.toString()
+            val password = CharArray(binding.passwordText.length())
+            binding.passwordText.text?.getChars(0, binding.passwordText.length(), password, 0)
+
+            val actionResult: SignInResult = authClient.signIn(
+                username = email,
+                password = password
+            )
+            binding.passwordText.text?.clear()
+            StringUtil.overwriteWithNull(password)
+
+            when (actionResult) {
+                is SignInResult.Complete -> {
+                    Toast.makeText(
+                        requireContext(),
+                        getString(R.string.sign_in_successful_message),
+                        Toast.LENGTH_SHORT
+                    ).show()
+                    displaySignedInState(accountState = actionResult.resultValue)
+                }
+                is SignInResult.MFARequired -> {
+                    val awaitingMFAState = actionResult.nextState
+                    val checkDefaultResult = awaitingMFAState.sendChallenge()
```

Comment:
let's call this something more meaningful than `checkDefaultResult`

Reply:
Updated it to requestDefaultAuthMethodResult